### DISCUSSION TITLE
Invalid left-hand side in assignment

### DISCUSSION THREAD

**Discussion by randalpinto** [vercel/next.js#29629] [Help] [answered]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

12.22.1

### What browser are you using?

Chrome

### What operating system are you using?

MacOS

### How are you deploying your application?

Other platform

### Describe the Bug

When packaging up my application for serverless deployment the following code:

https://github.com/vercel/next.js/blob/canary/packages/next/server/next-server.ts#L295

gets packaged up to:

true = JSON.stringify(true);

which results in the following exception at runtime:

```
"/var/task/chunks/602.js:429285",
        "            true = JSON.stringify(true);",
        "            ^^^^",
        "",
        "SyntaxError: Invalid left-hand side in assignment",
        "    at wrapSafe (internal/modules/cjs/loader.js:988:16)",
        "    at Module._compile (internal/modules/cjs/loader.js:1036:27)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)",
        "    at Module.load (internal/modules/cjs/loader.js:937:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:778:12)",
        "    at Module.require (internal/modules/cjs/loader.js:961:19)",
        "    at require (internal/modules/cjs/helpers.js:92:18)",
        "    at Object.__webpack_require__.f.require (/var/task/pages/api/test.js:495:28)",
        "    at /var/task/pages/api/test.js:418:40",
        "    at Array.reduce (<anonymous>)"
```

### Expected Behavior

No exception

### To Reproduce

I am deploying my app using: `https://github.com/serverless-nextjs/serverless-next.js`

A simple API route that returns 200 OK results in this error.

**Comment by PatrickChoDev** [comment]
isn't `true` the JavaScript and TypeScript restricted words?
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html

**Comment by ijjk** [comment]
Hi, this sounds like a bug in how the bundling in `serverless-next` is being handled as `next-server` shouldn't be processed with webpack which it sounds like it is since the `process.env` value is being replaced probably via `DefinePlugin` which shouldn't be run on `next-server`.

**Reply by randalpinto** [reply]
@ijjk the plot thickens, i have tried a completely different serverless deployment method (using AWS amplify) and i get the exact same error. Someone else in the serverless-next issue i opened said:

`i had similar, and it had to do with version conflict between next, webpack and typescript. what version of typescript and webpack are being used?`

I am using:

```
"typescript": "4.4.3",
"webpack": "5.56.0",
"next": "11.1.2",
```

Any ideas?

**Comment by randalpinto** [comment, marked as answer]
right, after a tortuous path I finally found the culprit. We use Sentry and deep in their docs it says that the webpack configuration is not compatible with serverless environments: https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/. This can be closed.

**Comment by zigang93** [comment]
after few hour of investigation, 
middleware.ts will broken when your next.config.js have webpack config like this below: 

remove the config will start your middleware.. 
@ijjk any other solution to have build ID ? I want notify user when new build was deploy to docker

```js
//  next.config.js
webpack: (config, { buildId }) => {
    config.plugins.push(
      new webpack.DefinePlugin({
        'process.env': {
          BUILD_ID: JSON.stringify(buildId),
        },
      })
    )
    return config
  },
```

**Reply by KosGrillis** [reply]
I can confirm that this is the culprit. Does any one have any fixes for this?

**Reply by favll** [reply]
Simply rewriting the key for the `DefinePlugin` was enough to fix the issue for me. 

```js
    config.plugins.push(
      new webpack.DefinePlugin({
        "process.env.BUILD_ID": JSON.stringify(buildId),
      })
    );
```


This is probably due to how the [`DefinePlugin`](https://webpack.js.org/plugins/define-plugin/) works. To quote the docs:

> When defining values for process prefer 'process.env.NODE_ENV': JSON.stringify('production') over process: { env: { NODE_ENV: JSON.stringify('production') } }. Using the latter will overwrite the process object which can break compatibility with some modules that expect other values on the process object to be defined.

**Reply by maxnowack** [reply]
Thanks, @favll! This has done the trick for me as well
